cdap-support-bundle/src/main/java/io/cdap/cdap/support/services/SupportBundleGenerator.java

```diff
@@ -167,13 +186,141 @@ void deleteOldFoldersIfExceedLimit(File baseDirectory) throws IOException {
     }
   }
 
+  /**
+   * Deletes select folder
+   */
+  public void deleteBundle(String uuid) throws IOException, NotFoundException {
+    File uuidFile = getUUIDFile(uuid);
+    if (!uuidFile.exists()) {
+      throw new NotFoundException(String.format("No such uuid '%s' in Support Bundle.", uuid));
+    }
+    DirUtils.deleteDirectoryContents(uuidFile);
+  }
+
+  /**
+   * Get single support bundle overall status
+   */
+  public SupportBundleOperationStatus getBundle(String uuid) throws IOException, NotFoundException {
+    File uuidFile = getUUIDFile(uuid);
+    if (!uuidFile.exists()) {
+      throw new NotFoundException(String.format("No such uuid '%s' in Support Bundle.", uuid));
+    }
+
+    File statusFile = new File(uuidFile, "status.json");
+    SupportBundleTaskReportStatus taskReportStatus = new SupportBundleTaskReportStatus(CollectionState.NOT_FOUND,
+                                                                                       CollectionState.NOT_FOUND,
+                                                                                       CollectionState.NOT_FOUND,
+                                                                                       CollectionState.NOT_FOUND,
+                                                                                       CollectionState.NOT_FOUND);
+    SupportBundleOperationStatus bundleOperationStatus = new SupportBundleOperationStatus(uuid,
+                                                                                          CollectionState.NOT_FOUND,
+                                                                                          taskReportStatus);
+    if (statusFile.exists()) {
+      SupportBundleStatus bundleStatus = getBundleStatus(uuidFile);
+      Set<SupportBundleTaskStatus> supportBundleTaskStatusSet = bundleStatus.getTasks();
+      taskReportStatus = collectSupportBundleTaskStatus(supportBundleTaskStatusSet);
+      bundleOperationStatus =
+        new SupportBundleOperationStatus(uuidFile.getName(), bundleStatus.getStatus(),
+                                         taskReportStatus);
+    }
+    return bundleOperationStatus;
+  }
+
+  /**
+   * Gather all absolute path file name list under certain uuid
+   */
+  public String createUuidZipFile(String uuid, Path tmpPath)
```

Comment:
i don't think predicate is useful since the bundle bundleRequestFileList already has every file we need to download, so I will just make both to use bundleRequestFileList